src/primitives.ts

```diff
@@ -930,180 +930,61 @@ export class Wire {
 }
 
 
-
-// NOUVELLES FONCTIONNALITES EN VUE D'UN REFACTOR
 export class PieChart {
-
-  point_list: Point2D[]=[];
   pieParts: Array<PiePart>=[];
-  all_attributes: Attribute[]=[];
-  lists: any[]=[];
-  to_display_attributes: Attribute[]=[];
+  attributeNames: Attribute[]=[]; // kept for a future release but useless now
 
   constructor(public tooltip:Tooltip,
-              public attribute_names:string[],
-              public point_style:PointStyle,
-              public surface_style:SurfaceStyle,
-              public edge_style:EdgeStyle,
-              public elements: any[],
-              public axis:Axis,
+              public slicingVariable:string,
+              public dataSamples: any[],
```

Comment:
```
public dataSamples : Map<string, any>
```

Reply:
put Array<Object> instead because of the deserialize method :
```
public static deserialize(serialized): PieChart { // TODO : clarify this
    return new PieChart(serialized['slicing_variable'],
                        serialized['data_samples'],
                        serialized['name']);
  }
```

where data_samples is a list of objects coming from python

Reply:
Done with this structure : 
```
export type DataSample = Map<string, any>
export class PieChart {
    public dataSamples: DataSample[],
```